src/gateway.rs

```diff
@@ -1681,89 +1668,47 @@ struct HeartbeatThreadCommunication {
     sequence_number: Option<u64>,
 }
 
-/**
-Trait which defines the behavior of an Observer. An Observer is an object which is subscribed to
-an Observable. The Observer is notified when the Observable's data changes.
-In this case, the Observable is a [`GatewayEvent`], which is a wrapper around a WebSocketEvent.
- */
-pub trait Observer<T: types::WebSocketEvent>: std::fmt::Debug {
-    fn update(&mut self, data: &T);
+/// Trait which defines the behavior of an Observer. An Observer is an object which is subscribed to
+/// an Observable. The Observer is notified when the Observable's data changes.
+/// In this case, the Observable is a [`GatewayEvent`], which is a wrapper around a WebSocketEvent.
+/// Note that `Debug` is used to tell `Observer`s apart when unsubscribing.
+pub trait Observer<T>: Sync + Send + std::fmt::Debug {
```

Comment:
Seems like you also already removed the WebSocketEvent restriction and made observers universal, hmmm